src/unused_finder/unused_finder.rs

```diff
@@ -0,0 +1,323 @@
+                return Err(napi::Error::new(napi::Status::InvalidArg, e.to_string()));
+            }
+        };
+        let skipped_items = Arc::new(skipped_items);
+        let mut flattened_walk_file_data =
+            create_flattened_walked_files(&paths_to_read, &skipped_dirs, &skipped_items);
+
+        let mut files: Vec<GraphFile> = flattened_walk_file_data
+            .par_iter_mut()
+            .map(|file| {
+                process_import_export_info(file, &resolver);
+                GraphFile::new(
+                    file.source_file_path.clone(),
+                    file.import_export_info
+                        .exported_ids
+                        .iter()
+                        .map(|e| e.metadata.export_kind.clone())
+                        .collect(),
+                    file.import_export_info.clone(),
+                    entry_packages.contains(&file.package_name), // mark files from entry_packages as used
+                )
+            })
+            .collect();
+
+        let files: HashMap<String, Arc<GraphFile>> = files
+            .par_drain(0..)
+            .map(|file| (file.file_path.clone(), Arc::new(file)))
+            .collect();
+
+        let graph = Graph { files };
+
+        let file_path_exported_items_map =
+            create_report_map_from_flattened_files(&flattened_walk_file_data);
+
+        Ok(Self {
+            config,
+            entry_packages,
+            tsconfig,
+            skipped_dirs,
+            skipped_items,
+            flattened_walk_file_data,
+            graph,
+            file_path_exported_items_map,
+            ..Default::default()
+        })
+    }
+
+    #[napi]
+    pub fn refresh_file_list(&mut self) {
+        let resolver = create_caching_resolver(&self.tsconfig);
+        self.flattened_walk_file_data = create_flattened_walked_files(
+            &self.config.paths_to_read,
+            &self.skipped_dirs,
+            &self.skipped_items,
+        );
+        self.file_path_exported_items_map = self
+            .flattened_walk_file_data
+            .clone()
+            .par_drain(0..)
+            .map(|file| {
+                let ids = file
+                    .import_export_info
+                    .exported_ids
+                    .iter()
+                    .map(|exported_item| ExportedItemReport {
+                        id: exported_item.metadata.export_kind.to_string(),
+                        start: exported_item.metadata.span.lo.to_usize() as i32,
+                        end: exported_item.metadata.span.hi.to_usize() as i32,
+                    })
+                    .collect();
+                (file.source_file_path.clone(), ids)
+            })
+            .collect();
+        self.flattened_walk_file_data
+            .par_iter_mut()
+            .for_each(|file| {
+                process_import_export_info(file, &resolver);
+            });
+
+        self.refresh_graph();
+    }
+
+    fn refresh_graph(&mut self) {
```

Comment:
there isn't a structural throughline in what these refresh_* methods are doing.
From what I can tell:

- refresh_file_list re-scans the entire project from disk by re-walking the filesystem
- refresh_files_to_check does a partial re-build of already seen edited files in the files list
- refresh_graph is a helper that rebuilds the internal import graph data structure from mutations made during refresh_file_list or refresh_files_to_check